shared/route-tree/index.js

```diff
@@ -0,0 +1,149 @@
+// @flow
+import * as I from 'immutable'
```

Comment:
🐑  minor thing: I have a slight preference for `import * Immutable from 'immutable'` so that it's easier to grep for in our code.